data/tutorials/language/0it_06_hof.md

```diff
@@ -0,0 +1,930 @@
+# Higher-Order Functions
+
+**Prerequisites:**
+* Functions
+* Recursion
+
+**What you'll learn**:
+* What is a higher-order function
+* What are some common higher-order functions
+* What is function composition
+* How to compose functions
+* How to make reusable functions
+* How to make functions fit anywhere
+
+**(INTERNAL): journey:**
+* introduction – motivate the existence of higher-order functions with an example
+* how do i make new functions from existing functions?
+* composing pipelines and `|>`
+    * t-first vs t-last
+* point-free composition
+    * how we can do it
+    * why we don't do it
+* usual gotchas / patterns
+    * i need a function that does nothing: Fun.id
+    * my function is backwards: flip
+    * my function takes too many params:
+        * if you can, name it
+        * partial application in pipelines
+        * they're in the wrong order
+            * lambda wrappers (fun x -> f 123 x true)
```

Comment:
These are internal notes to illustrate a little bit of the learning journey I had in mind (as we discussed back in December). Feel free to remove this.

Reply:
Hi! Internal notes meaning they won't be visible on the page?

Reply:
Yup! Just here for the review process.